### DISCUSSION TITLE
Improve the view of the bbdd

### DISCUSSION THREAD

**Discussion by TecnoTalarn** [foldynl/QLog#378] [Ideas]
When we enter a call sign, the database shows us the times we have worked on it. But if we have never worked on it, the database shows nothing.
I think it would be good if the correspondent had not previously worked, to show us the last contacts with the entity we are working with.
For example, if I'm talking to VK8AA a call sign that I've never worked before, it will show me all the VKs in the list.

Like this image
![image](https://github.com/foldynl/QLog/assets/61935185/12aed720-3c02-45e4-9264-87d62f9471f0)

**Comment by foldynl** [comment]
I have to think about this. The original meaning was that it should find all previous QSOs with the same callsign so that you can see the "history". However, your idea has a good basis, because you would have seen QSOs for the given country. 

I opened an issue #381 for tracking the change.

**Comment by TecnoTalarn** [comment]
Yes, the idea of ​​seeing the story with the correspondent is correct, and I already like it, In addition, it is important to be able to see it, while you are talking with him.
My proposal is that if there is no history with the correspondent, that you can see the history with the correspondent's entity. So we have more information at hand, and not just the DXCC information that only gives us the band and the mode, but not the date, or the callsigns.

**Comment by foldynl** [comment]
Even though I initially thought it was a good idea, I tried to implement it and, as a user, I don't like it very much. This is because, when I am writing, I expect the entered callsign to be searched. This can happen if you have the first characters of the callsign, but then when the callsign (or part of the callsign) is not found, it unexpectedly switches to searching by prefix. In my opinion, it is confusing for the user when the first characters are searched for a partial match of the callsign, and then it suddenly switches to an exact match of the prefix.

Therefore, I will not implement it now but leave it open. I need to find some visual recognition between the partial match of the callsign and the exact match of the prefix.

**Comment by TecnoTalarn** [comment]
Well, it was an idea. it might be necessary to give it another go and see how to visualize the result so that it is not a sudden change. I don't know if it's easy to do, and possibly there are other more urgent things.

Thanks anyway for trying it.

**Reply by foldynl** [reply]
It is definitely a legitimate request. However, I need to find a suitable way to inform the user that the search criteria change when the callsign is typed. Sometimes, it takes me a long time to find a solution.
